lp1742617: Don't import/read custom comment tags

uklotzde opened the pull request:
Fixes a minor metadata bug:

https://bugs.launchpad.net/mixxx/+bug/1742617

* Added a regression test for ID3v2 tags
* Functionality reused for VorbisComment tags that we parse like MusicBrainz suggests
* The import of RIFF tags contained redundant code -> removed

uklotzde committed:
Omit reading the default comment field on demand

uklotzde committed:
Ignore default comment field when raeading ID3v2 and VorbisComment tags

uklotzde committed:
Reuse existing code when reading RIFF tags

uklotzde committed:
Add regression test for custom comment tags

uklotzde committed:
Merge branch '2.1' into lp1742617_filter_comment_tags

daschuer commented on src/track/trackmetadatataglib.cpp:
```diff
@@ -1408,13 +1415,14 @@ void importTrackMetadataFromAPETag(TrackMetadata* pTrackMetadata, const TagLib::
     }
 
-    importTrackMetadataFromTag(pTrackMetadata, tag);
+    importTrackMetadataFromTag(pTrackMetadata, tag, READ_TAG_OMIT_COMMENT);
```
Please add a comment, why we omit comments here.

uklotzde committed:
Clarify handling of comment tags

uklotzde committed:
Add special case handling for null strings

...otherwise comments tags with an empty non-null description string
will be written.

daschuer commented:
Thank you :-) LGTM!
